src/main/java/mongodb/Main.java

```diff
@@ -0,0 +1,220 @@
+    // changes we just made.
+    try {
+      Recipe updatedDocument = collection.findOneAndUpdate(findPotato,
+              updateFilter, options);
+      if (updatedDocument == null) {
+        System.out.println("Couldn't update the recipe.");
+      } else {
+        System.out.println("Updated the recipe to: " + updatedDocument);
+      }
+    } catch (MongoException me) {
+      System.err.println("Could not update any recipes due to an error: " + me);
+    }
+
+    /*      *** DELETE DOCUMENTS ***
+     *
+     *      As with other CRUD methods, you can delete a single document
+     *      or all documents that match a specified filter. To delete all
+     *      of the documents in a collection, pass an empty filter to
+     *      the deleteMany() method. In this example, we'll delete 2 of
+     *      the recipes.
+     */
+    Bson deleteFilter = Filters.in("name", Arrays.asList("elotes", "fried rice"));
+    try {
+      DeleteResult deleteResult = collection
+              .deleteMany(deleteFilter);
+        System.out.println(String.format("Deleted %d documents.", deleteResult.getDeletedCount()));
+    } catch (MongoException me) {
+      System.err.println("Could not delete any recipes due to an error: " + me);
+    }
+
+    // always close the connection when done working with the client
+    mongoClient.close();
+  }
+
+  // helper method with a few pre-wired recipes we can insert into the database as examples.
+  public static List<Recipe> getRecipes() {
+    return Arrays.asList(
+            new Recipe("elotes",
+                    Arrays.asList("corn", "mayonnaise", "cotija cheese", "sour cream", "lime" ),
+                    35),
+            new Recipe("loco moco",
+                    Arrays.asList("ground beef", "butter", "onion", "egg", "bread bun", "mushrooms" ),
+                    54),
+            new Recipe("patatas bravas",
+                    Arrays.asList("potato", "tomato", "olive oil", "onion", "garlic", "paprika" ),
+                    80),
+            new Recipe("fried rice",
+                    Arrays.asList("rice", "soy sauce", "egg", "onion", "pea", "carrot", "sesame oil" ),
+                    40)
+    );
+  }
+
+  // POJO (Plain Old Java Object) class defining a recipe. This class is a POJO because it contains getters and
+  // setters for every member variable as well as an empty constructor.
+  static public class Recipe {
+    private String name;
+    private List<String> ingredients;
+    private int prepTimeInMinutes;
+
+    public Recipe(String name, List<String> ingredients, int prepTimeInMinutes) {
+      this.name = name;
+      this.ingredients = ingredients;
+      this.prepTimeInMinutes = prepTimeInMinutes;
+    }
+
+    // empty constructor required when we fetch data from the database -- getters and setters are later used to
+    // set values for member variables
+    public Recipe() {
+      ingredients = new ArrayList<>();
```

Comment:
Question: more for personal edification, but do you need to initialize this? If so, why not provide the type of the ArrayList? And why not the `name` field (String object)?

Reply:
I could be wrong, but I believe that leaving a null member in place after calling a constructor with no arguments is frowned upon. It would be pretty easy to create an object this way, then try to add something to the list of ingredients and end up with a good ol' null pointer exception. Same log applies to mutating the name, too -- I'll add that for consistency.

These are definitely not necessary, but especially when we're dealing with folks who might not know best practices I feel like it's safer to err on the cautious side of best practices.